WordPress/wporg-main-2022
# Try new Local Navigation block

This is a companion PR to https://github.com/WordPress/wporg-mu-plugins/pull/404, which introduces a new Local Navigation container to hold the section title & section menu. This PR extracts out the local header from the synced post content, and creates new reusable pattern-partials which handle the header content on each subpage. The new patterns use the Local Navigation container block.

The changes here also bring these headers in line with the generic section header mocked up. @WordPress/meta-design As I understood it, this pattern is complete, but if that's not the case let me know.

![Section Header L2](https://github.com/WordPress/wporg-main-2022/assets/541093/f8638edf-41a4-4f00-922b-2cf4957f780a)

❗ If we decide not to merge the Local Nav block, I still think we should merge https://github.com/WordPress/wporg-main-2022/commit/e1e42a3e60411e69d1f81841cb1ef0b61f31691c to DRY up the About pages. That won't change any visuals.

Additionally, once this is merged, I'll need to go in and update each of the subpages to remove the banner, so that the content sync doesn't overwrite these changes.

See WordPress/wporg-mu-plugins#465

### Screenshots

1. The Stats page, a general About subpage. I added borders to the header as this was mocked up at some point, but I don't know if that's still the preference.
2. The Requirements page, scrolled down— the top banner sticks under the global header, but the breadcrumbs scroll. Is that the expected behavior?
3. The Beta/Nightly page— I kept the brush stroke style on the download subpages, but that + the breadcrumbs is a little strange.
4. The download counter— unlike the other download subpages, this does not have the breadcrumbs, and uses the black+borders style for the banner. So we can still have unique features on individual pages.

| Before | After |
|--------|-------|
| ![before-about-stats](https://github.com/WordPress/wporg-main-2022/assets/541093/82ae7057-f513-4b95-ac90-83d6f2e4bab7) | ![after-about-stats](https://github.com/WordPress/wporg-main-2022/assets/541093/d8ecd2c8-242b-4135-9690-3448d79c2640) |
| ![before-about-reqs-scroll](https://github.com/WordPress/wporg-main-2022/assets/541093/0fc0077d-4125-4027-a1e7-b37206c3828b) | ![after-about-reqs-scroll](https://github.com/WordPress/wporg-main-2022/assets/541093/b31df9d1-6443-4c7b-b2fe-91b477494270) |
| ![before-download-nightly](https://github.com/WordPress/wporg-main-2022/assets/541093/1983fcfb-f6d4-4fae-8b20-668e8a8fcca3) | ![after-download-nightly](https://github.com/WordPress/wporg-main-2022/assets/541093/f068f201-a06c-444c-a9ef-db998f530fe5) |
| ![before-download-counter](https://github.com/WordPress/wporg-main-2022/assets/541093/6578d80a-b373-41c6-9c0c-8a830be09319) | ![after-download-counter](https://github.com/WordPress/wporg-main-2022/assets/541093/a1b7563a-1878-4fbf-80ac-3f3dd75e4997) |

### How to test the changes in this Pull Request:

1. Make sure you have https://github.com/WordPress/wporg-mu-plugins/pull/404 applied on wporg-mu-plugins
2. Check out this branch
3. View the About and Download subpages
